### PR TITLE
Add bootstrap-year-calendar w/ npm auto-update via single package.json

### LINKED ISSUE #10120: [Request] Add bootstrap-year-calendar
**Library name:** bootstrap-year-calendar
**Git repository url:** https://github.com/Paul-DS/bootstrap-year-calendar
**npm package url(optional):**  bootstrap-year-calendar
**License(s):** Apache-2.0
**Official homepage:** http://www.bootstrap-year-calendar.com/

For beginners who is interested in picking up this issue:

In this issue, we are going to host a new web front-end library on CDNJS, so that web developer can easily use it on our free CDN without additional download/upload process 🚀 .

If you'll like to work locally, I'll suggest you start with sparseCheckout: https://github.com/cdnjs/cdnjs/blob/master/documents/sparseCheckout.md, since there are too many files in this repo, without sparseCheckout, the whole process will be super slow, using git command line can learn more skills and experience, if you don't like that, instead, you can also just do it on GitHub, it's more convenient, but it may be a little bit harder to fix problems on GitHub gui.

Here are the docs about how to add a library:
 - https://github.com/cdnjs/cdnjs#adding-a-new-or-updating-an-existing-library
 - https://github.com/cdnjs/cdnjs/blob/master/.github/CONTRIBUTING.md#d-adding-a-new-library

We'll also ask to setup auto-update config so that the added libraries can be updated to the latest version easily:
 - https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md

For beginners who is interested in join and help add this library, please feel free to tell me what kind of help you'll need, we'll help you send the valid pull request, thanks 😄 .

### PR DESCRIPTION
**Pull request by dakshshah96**
close #10120, cc @Paul-DS

Pull request for issue: #10120

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/Paul-DS/bootstrap-year-calendar
 * Official website (optional, not the repository): http://www.bootstrap-year-calendar.com/
 * NPM package url (optional): https://www.npmjs.com/package/bootstrap-year-calendar
 * License and its reference: Apache-2.0
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 18
   - Count of stars: 131
   - Count of forks: 62
 * NPM download stats (optional):
   - Downloads in the last day: 3
   - Downloads in the last week: 226
   - Downloads in the last month: 724

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/bootstrap-year-calendar/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "bootstrap-year-calendar",
+  "description": "A fully customizable year calendar widget, for Bootstrap!",
+  "homepage": "http://www.bootstrap-year-calendar.com/",
```
Comment (pvnr0082t) [inline ajax/libs/bootstrap-year-calendar/package.json]: It's better to use `https` instead `http` here.

File: ajax/libs/bootstrap-year-calendar/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/Paul-DS/bootstrap-year-calendar"
+  },
+  "author": "Paul-DS <@Paul-DS> (https://taylorhakes.com/)",
+  "license": "Apache-2.0",
+  "filename": "bootstrap-year-calendar.js",
```
Comment (pvnr0082t) [inline ajax/libs/bootstrap-year-calendar/package.json]: `filename` should point to the main minified file, `js/bootstrap-year-calendar.min.js`.

File: ajax/libs/bootstrap-year-calendar/package.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      "files": [
+        "bootstrap-year-calendar*"
+      ]
+    },
+    {
+      "basePath": "css",
+      "files": [
+        "bootstrap-year-calendar*"
+      ]
+    }
```
Comment (pvnr0082t) [inline ajax/libs/bootstrap-year-calendar/package.json]: It's not good to use `js` and `css` as basePath. Let's leave basePath blank and use following as the rule:
```
    {
      "basePath": "",
      "files": [
        "+(js|css)/**/bootstrap-year-calendar*"
      ]
    }
```
Thanks.

**Comment by dakshshah96** [comment]
@pvnr0082t Thanks! Yes, I could have done that. Was waiting for changes to be approved, then I was planning to squash commits and rebase. 👍

File: ajax/libs/bootstrap-year-calendar/package.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  "npmName": "bootstrap-year-calendar",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "+(js|css)/**/bootstrap-year-calendar*"
+      ]
+    }
+  ]
+}
```
Comment (dakshshah96) [inline ajax/libs/bootstrap-year-calendar/package.json]: @PeterDaveHello How do I use `fixFormat.js` for my local respository? Need some help with that.

Comment (maruilian11) [inline ajax/libs/bootstrap-year-calendar/package.json]: @dakshshah96 just use `./tools/fixFormat.js` in `cdnjs` root. :)

**Pull request merged**